src/components/Accordion/Accordion.tsx

```diff
@@ -8,12 +8,59 @@ import AccordionContent from './AccordionContent'
 import { DefaultBehavior } from '../../lib/accessibility'
 import { Accessibility } from '../../lib/accessibility/interfaces'
 
+export interface IAccordionPropsStrict {
+  /** An element type to render as (string or function). */
+  as?: any
+
+  /** Index of the currently active panel. */
+  activeIndex?: number[] | number
+
+  /** Primary content. */
+  children?: React.ReactNode
+
+  /** Additional classes. */
+  className?: string
+
+  /** Initial activeIndex value. */
+  defaultActiveIndex?: number[] | number
+
+  /** Only allow one panel open at a time. */
+  exclusive?: boolean
+
+  /**
+   * Called when a panel title is clicked.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All item props.
+   */
+  onTitleClick?: Function
+
+  /** Shorthand array of props for Accordion. */
+  panels?: {
+    content: React.ReactNode | object
+    title: React.ReactNode | object
+  }[]
+
+  /** Accessibility behavior if overridden by the user. */
+  accessibility?: object | Function
+
+  /** Custom styles to be applied for component. */
+  styles?: object | Function
```

Comment:
Actually this would be the following form {root: () => ICSSInJSStyle, otherPart: () => ICSSInJSStyle...} so we cannot generalize to just ICSSInJSStyle

Reply:
Talking about styles, we may be able to define the variables and styles in the following way:
variables?: ComponentVariablesInput
styles?: IComponentPartStylesInput
as these are the types used in the renderComponent. @levithomason am I right about this?